audio_play/src/audio_play.cpp

```diff
@@ -43,13 +45,15 @@ namespace audio_transport
           _convert = gst_element_factory_make("audioconvert", "convert");
           audiopad = gst_element_get_static_pad(_convert, "sink");
           _sink = gst_element_factory_make("autoaudiosink", "sink");
+          if (!device.empty()) {
+            g_object_set(G_OBJECT(_sink), "device", device.c_str(), NULL);
```

Comment:
This "device" property does not exist for `autoaudiosink`, you would need to change it to `alsasink` or it may prefer something besides alsa.